### PR TITLE
Clarified action of function and added example

### PR DESCRIPTION
**Pull request by jimmonte**
The example shows how stopPropagation and stopImmediatePropagation work.

### CODE REVIEW

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 1]
```diff
@@ -13,14 +13,108 @@ called.
...
 event.stopImmediatePropagation()
 ```
 
+### Comparing event stopping functions
+
+The example below has three buttons inside of three nested divs. Each button has two
+event listeners registered for click events, and each div has an event listener, also
+registered for click events. The top button allows normal event propagation while the
+middle button calls `stopPropagation()` in its first callback function and the bottom
+button calls `stopImmediatePropagation()` in its first callback function.
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:31]:
```suggestion
The example below has three buttons inside three nested {{HTMLElement("div")}} elements. Each button has two
event listeners registered for click events, and each `<div>` has an event listener, also
registered for click events.

- The top button allows normal event propagation.
- The middle button calls `stopPropagation()` in its first callback function
- The bottom button calls `stopImmediatePropagation()` in its first callback function.
```

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 2]
```diff
@@ -13,14 +13,108 @@ called.
...
+is invoked during one such call, no remaining listeners will be called, either on that
+element or any other element.
 
 ## Syntax
 
 ```js-nolint
 event.stopImmediatePropagation()
 ```
 
+### Comparing event stopping functions
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:25]: We have quite a particular structure for reference pages: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_method_subpage_template.

Especially, examples should go inside their own "Examples" H2, and also see the formatting we use for live samples: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#traditional_live_samples.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 3]
```diff
@@ -13,14 +13,108 @@ called.
...
+  <script>
+    const outElem = document.getElementById("output");
+
+    /* Clear the output */
+    document.addEventListener("click", () => outElem.innerHTML = "", true);
+
+    [1, 2, 3].forEach(value => {
+      const id = "div" + value;
+      document.getElementById(id).addEventListener("click", () =>
+          outElem.innerHTML += "click event processed on " + id + "<br>"
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:52]: We should not write to `innerHTML`. In cases like this I usually make the output element a `<pre>` and set its `textContent` (e.g. https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/progress_event#live_example).

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 4]
```diff
@@ -13,14 +13,108 @@ called.
...
+  <div id="output"></div>
+  <script>
+    const outElem = document.getElementById("output");
+
+    /* Clear the output */
+    document.addEventListener("click", () => outElem.innerHTML = "", true);
+
+    [1, 2, 3].forEach(value => {
+      const id = "div" + value;
+      document.getElementById(id).addEventListener("click", () =>
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:51]: I guess this is a matter of personal taste but I would far rather repeat the `addEventListener` calls here.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 5]
```diff
@@ -13,14 +13,108 @@ called.
...
+
+```html
+  <h2>Click on the buttons</h2>
+  <div id="div1">div1<br><div id="div2">div2<br><div id="div3">div3<br>
+    <button id="allow">allow propagation</button><br>
+    <button id="stop">stop propagation</button><br>
+    <button id="istop">immediate stop propagation</button>
+  </div></div></div>
+  <div id="output"></div>
+  <script>
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:43]: This should be in a separate `### JavaScript` section, per https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#traditional_live_samples.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 6]
```diff
@@ -13,14 +13,108 @@ called.
...
+button calls `stopImmediatePropagation()` in its first callback function.
+
+#### HTML
+
+```html
+  <h2>Click on the buttons</h2>
+  <div id="div1">div1<br><div id="div2">div2<br><div id="div3">div3<br>
+    <button id="allow">allow propagation</button><br>
+    <button id="stop">stop propagation</button><br>
+    <button id="istop">immediate stop propagation</button>
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:40]: nit, but `"immediate-stop"` would be better than `"istop"`

File: files/en-us/web/api/event/stopimmediatepropagation/index.md (hunk 3 above)
Comment (wbamberg) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md:52]: We also prefer template literals to `+` for concatenation: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#template_literals.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 7]
```diff
@@ -13,14 +13,108 @@ called.
...
+      width: 100px;
+      color: #008;
+      padding: 5px;
+      margin: 5px;
+    }
+```
+
+#### Result
+
+The click callback functions display messages when they are called.
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:111]:
```suggestion
The click event handlers display messages when they are called.
```

File: files/en-us/web/api/event/stopimmediatepropagation/index.md [hunk 8]
```diff
@@ -13,14 +13,108 @@ called.
...
+It is seen that calling `stopPropagation()` allows the other callback
+function registered for clicks on the button to execute while calling
+`stopImmediatePropagation()` does not.
```
Comment (wbamberg) [inline files/en-us/web/api/event/stopimmediatepropagation/index.md:114]:
```suggestion
You will see that calling `stopPropagation()` allows the other event handlers registered for clicks on the button to execute while calling `stopImmediatePropagation()` does not.
```

File: files/en-us/web/api/event/stopimmediatepropagation/index.md (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md:31]: We don't really encourage hard line breaks for prose either, although I wouldn't reject a PR for it.

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thank you @jimmonte , this is a very nice example. Most of my comments are about formatting.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md (hunk 4 above)
Comment (jimmonte) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md:51]: I think I have a more agreeable alternative that is still DRY.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md (hunk 3 above)
Comment (jimmonte) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md:52]: I had used innerHTML due to the <br> nodes, but <pre> actually allows some simplification besides getting rid of the innerHTML.

Comment (jimmonte) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md:52]: I had used innerHTML due to the `<br>` nodes, but `<pre>` actually allows some simplification besides getting rid of the innerHTML.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md (hunk 5 above)
Comment (wbamberg) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md:43]: You can rely on the DOM being available.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md (hunk 8 above)
Comment (jimmonte) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md]: I resolved all of the issues and made the example a bit prettier. I see the commit I made, and I tried to create a new pull request. However, I am questioning if the pull request exists. Is there a new pull request?

Comment (wbamberg) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md]: No (as you probably saw by now) your new changes get added to this pull request. Which is what we want :).

Sorry I've not had a chance to review this today but I will tomorrow.

**Comment by wbamberg** [comment]
@jimmonte , I appreciate the work you've put into this and I don't want to be discouraging, but honestly I think your original example was more effective, as an example, than this one.

With any example, there's the actual code we are showcasing (in this case `stopImmediatePropagation` and `stopPropagation`) and there's all the framework code that's needed to built it into an executable example that makes a tangible difference. Every bit of complexity in the framework code makes the example harder for people to understand, and makes more cognitive demands on them.

So I think examples should be as simple as they can, while still showing the effect of the code being showcased. I don't think (for example) animations are worth the complexity they add to the example.

There's also a lot of abstraction in this example, which makes it harder for people to understand. For example, `addElementAction` is quite an abstracted function, with quite a lot of complexity that's not relevant to the code lesson of the example. And the abstraction adds complexity (for example, 4 arguments). DRY has merits but often for examples it's clearer to have less abstraction even at the code of a little repetition.

**Comment by jimmonte** [comment]
@wbamberg Thanks for the feedback. No hard feelings at all. You are supposed to provide a review that will improve the document. I will simplify the example and give an update.

File: files/en-us/web/api/event/stopimmediatepropagation/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/api/event/stopimmediatepropagation/index.md]: I think the only "rule" is that we stick to rigidly is no one likes line breaks inside macros or within HTML tags inside HTML tables. 

FWIW I'm prefer semantically useful line breaks to arbitrary fixed-length lines. I'm in the minority that I like these breaks on sentences, while most of the docs either break on paragraphs or at random points.

**Comment by wbamberg** [comment]
Sorry to be so slow to review you PR @jimmonte . I haven't forgotten about it and will get to it as soon as I can.

**Comment by jimmonte** [comment]
No problem. I had only requested the review because I found the page for pending pull requests today, and it appeared that the process was being blocked by me in this case while the others were waiting for the reviewers, albeit for quite a while for some of them.
2023-04-05 14_37_06-Pull Requests

**Review by wbamberg — APPROVED** [review]
👍 looks great. Thank you for this very nice example, @jimmonte , and once again sorry for being slow.

**Pull request merged**